### PR TITLE
Refine CLI for use with `get_table` as `almanack <repo path>`

### LINKED ISSUE #134: Enable CLI use of `almanack.table(repo_url="")` to run `$ almanack <repo location here>`
The CLI currently only enables the use of the entropy measurements. We could expand this by adding integration with the `almanack.table()` functionality to provide more measurements.

May coincide with work in #82 .

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR refines the CLI to use `get_table` as `$ almanack <repo path>` when the Software Gardening Almanack is used through `python-fire`. The goal was to make using the internal functionality easily from outside of a Pythonic API, for example, as with workflow pipelines that may not use Python API's.

Thanks for any feedback! 

Closes #134 

## What is the nature of your change?

- [ ] Content additions or updates (adds or updates content)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own contributions.
- [x] I have commented my content, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to related documentation (outside of book content).
- [x] My changes generate no new warnings.
- [x] New and existing tests pass locally with my changes.
- [x] I have added tests that prove my additions are effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: src/almanack/cli.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+"""
+Setup Entropy Report CLI through python-fire
```
Comment (gwaybio) [inline src/almanack/cli.py:2]: It seems like this isn't restricted to just the entropy report?

File: tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+def test_cli_util():
+    """
+    Test the run_cli_command for successful output
+    """
+
+    command = """echo 'hello world'"""
+    _, _, returncode = run_cli_command(command)
+
+    assert returncode == 0
```
Comment (gwaybio) [inline tests/test_cli.py:23]: is there ever a need for the output to not be successful?

File: tests/utils.py [hunk 3]
```diff
@@ -17,3 +18,20 @@ def check_subproc_run_for_nonzero(completed_proc: subprocess.CompletedProcess) -
...
+
+    Args:
+        command (list): The command to run as a list of strings.
+
+    Returns:
+        tuple: (str: stdout, str: stderr, int: returncode)
+    """
+
+    result = subprocess.run(
+        command.split(" "), capture_output=True, text=True, check=False
```
Comment (gwaybio) [inline tests/utils.py:35]: It's unclear to me where/who `run_cli_command()` runs, but it feels a bit dangerous to split by a space always and by default

Comment (falquaddoomi) [reply tests/utils.py:35]: I have to agree with @gwaybio; there are situations, e.g. quoted strings, where you'd want to not tokenize the string on just spaces. One option is to use [shlex.split](https://docs.python.org/3.13/library/shlex.html#shlex.split), which will split a string into tokens as a shell would.

File: src/almanack/cli.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+if __name__ == "__main__":
+    """
+    Setup the CLI with python-fire for the almanack package.
+
+    This allows the function `check` to be ran through the
+    command line interface.
+    """
+    cli_get_table()
```
Comment (falquaddoomi) [inline src/almanack/cli.py:48]: I looked into this a bit, and it seems if your method returns a list it outputs each element of that list with newlines in between, which IMO constitutes distinction between the dicts, but still, I agree that it's a little odd. You can override this behavior when executing `fire.Fire()` by including the `serialize` argument, e.g.:

```suggestion
import json

import fire

from .metrics.data import get_table

def cli_get_table() -> None:
    """
    Creates Fire CLI for get_table_json_wrapper.

    This enables the use of CLI such as:
    `almanack <repo path>`
    """
    fire.Fire(get_table, serialize=json.dumps)


if __name__ == "__main__":
    """
    Setup the CLI with python-fire for the almanack package.

    This allows the function `check` to be ran through the
    command line interface.
    """
    cli_get_table()
```


*(I'm not sure why the diff shows my change as replacing the entire file, but I just deleted the wrapper function and fixed a typo in "almanack" in the `cli_get_table` docstring.)*

File: src/almanack/metrics/data.py [hunk 5]
```diff
@@ -20,7 +20,7 @@
 METRICS_TABLE = f"{pathlib.Path(__file__).parent!s}/metrics.yml"
 
 
-def get_table(repo_path: str) -> Dict[str, Any]:
+def get_table(repo_path: str) -> List[Dict[str, Any]]:
```
Comment (falquaddoomi) [inline src/almanack/metrics/data.py:23]: Nice catch!

**Review by falquaddoomi — COMMENTED** [review]
Nice work; seems like you cleaned up a lot of things and added some tests. I made a few comments which probably should be reviewed before merging, but nothing serious.

File: src/almanack/metrics/data.py (hunk 5 above)
Comment (d33bs) [reply src/almanack/metrics/data.py:23]: Cheers, thanks!

File: src/almanack/cli.py (hunk 4 above)
Comment (d33bs) [reply src/almanack/cli.py:48]: Thanks for this! I'll add this in as an additional change.

File: tests/utils.py (hunk 3 above)
Comment (d33bs) [reply tests/utils.py:35]: Thanks - I'll go ahead and make a change here to address the uncertainty here.

**Comment by d33bs** [comment]
Thanks @falquaddoomi for the review! I've made some changes based on your comments and revisited what @gwaybio mentioned, regarding the unsafe space splitting for the subprocess testing utility function, based on both of your feedback.

**Comment by d33bs** [comment]
Openly, I wondered if the default behavior of the output returned by `almanack <repo path>` should be a human-readable table instead of data. We could add a `--json` flag to return what is currently provided. Perhaps this is an additional issue/ PR at a later time?

**Comment by falquaddoomi** [comment]
> Openly, I wondered if the default behavior of the output returned by `almanack <repo path>` should be a human-readable table instead of data. We could add a `--json` flag to return what is currently provided. Perhaps this is an additional issue/ PR at a later time?

I think that sounds like a reasonable idea, if you expect people to be using it interactively rather than parsing it. I think it pertains to this PR so it would make sense to include it here, but it's up to you if you'd prefer for it to be reviewed in more detail in a different PR.

**Comment by d33bs** [comment]
Thanks @falquaddoomi - I created #172 to further this development. I don't know of an exact demand for this right now whereas we can more readily make use of this PR's CLI modifications as is.

Separately, @gwaybio , @falquaddoomi , is there anything else I may add to this PR in order to ready it for a merge? Thanks in advance for any further feedback!

**Review by falquaddoomi — APPROVED** [review]
I don't have anything to add; this all looks good to me!

**Pull request merged**